src/main/java/core/basesyntax/serviceimpl/ShopServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.operation.OperationHandler;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactionList) {
+        for (FruitTransaction transaction : transactionList) {
+            OperationHandler operationHandler =
+                    operationStrategy.get(transaction.getOperation());
+            operationHandler.operationType(transaction.getFruit(),
+                    transaction.getQuantity());
+        }
```

Comment:
The process method currently handles the loop and the application of strategies. To further adhere to the SOLID principles, consider separating the loop that goes through transactions from the execution of business logic which should be encapsulated in a separate method or class.